lib/core/engine/iteration.js

```diff
@@ -195,6 +198,21 @@ class Iteration {
               result[i].browserScripts.pageinfo.visualElements
             );
             result[i].visualMetrics = videoMetrics.visualMetrics;
+
+            const profileFilename = path.join(
+              this.engineDelegate.baseDir,
+              pathToFolder(result[i].url, options),
+              `geckoProfile-${index}.json`
+            );
+
+            try {
+              const geckoProfile = JSON.parse(fs.readFileSync(profileFilename));
+              geckoProfile.meta.visualMetrics = result[i].visualMetrics;
+              fs.writeFileSync(profileFilename, JSON.stringify(geckoProfile));
+            } catch (e) {
+              log.error('Adding visual metrics to Gecko profile failed', e);
```

Comment:
Be explicit about what was tried: include the filename in the error message, please.